### PR TITLE
Rolling provider

### PR DESCRIPTION
**Pull request by dannoh**
Might have gone a bit overboard here but was looking for a way to implement the game display logic I would like to see, and obviously maintain the existing way people are using already.
So, having seen the weather provider stuff MM is using now, I decided to implement something similar.  
All of the existing logic I moved into standardnhl.js, my preferred method is in rollingnhl.js

Another thing, the locale wasn't working for me, I didn't see anything about locale being a config value in MM so I wasn't sure if it was meant to be this.config instead.

### CODE REVIEW

File: MMM-NHL.js [hunk 1]
```diff
@@ -217,15 +217,15 @@ Module.register('MMM-NHL', {
         this.nunjucksEnvironment().addFilter('formatStartDate', game => {
             const now = new Date();
             const inAWeek = now.setDate(now.getDate() + 7);
-            const start = new Date(game.timestamp);
+            const start = new Date(game.gameDate);
 
             if (start > inAWeek) {
-                return new Intl.DateTimeFormat(config.locale, {
+                return new Intl.DateTimeFormat(this.config.locale, {
```
Comment (dannoh) [inline MMM-NHL.js:223]: This is what I meant by the locale issue.  Not sure if its correct or not, but I made the change and added locale to the MMM-NHL config and now the dates display correctly.

**Comment by fewieden** [comment]
@dannoh you put a lot of work in here, besides some problems that I can see in the code I think it is the wrong approach. What you want is a new config option that would apply to every data source as it is about what you want to display. You would add something similar to this if you want to integrate a new data source like ESPN for example. I think we can achieve this simpler.

PS: I released the current version of develop as v2 before we create an even bigger monster.

**Comment by fewieden** [comment]
Around 2 years ago I started implementing different data sources in one of my modules https://github.com/fewieden/MMM-Fuel/tree/master/apis they all provide the data in the same format and the module handles then all visualization instead of duplicating the same logic everywhere.

**Comment by fewieden** [comment]
I implemented the `rollOver` config option here https://github.com/fewieden/MMM-NHL/pull/17 what do you think about it?

File: MMM-NHL.js (hunk 1 above)
Comment (dannoh) [reply MMM-NHL.js:223]: Ok that makes sense, I missed that in the docs.  Will revert.

**Pull request closed without merging**